llvm/test/CodeGen/RISCV/double-convert.ll

```diff
@@ -734,38 +734,42 @@ define i64 @fcvt_l_d_sat(double %a) nounwind {
 ; RV32IZFINXZDINX-NEXT:    sw ra, 12(sp) # 4-byte Folded Spill
 ; RV32IZFINXZDINX-NEXT:    sw s0, 8(sp) # 4-byte Folded Spill
 ; RV32IZFINXZDINX-NEXT:    sw s1, 4(sp) # 4-byte Folded Spill
+; RV32IZFINXZDINX-NEXT:    sw s2, 0(sp) # 4-byte Folded Spill
+; RV32IZFINXZDINX-NEXT:    lui a2, %hi(.LCPI12_0)
+; RV32IZFINXZDINX-NEXT:    lw a4, %lo(.LCPI12_0)(a2)
+; RV32IZFINXZDINX-NEXT:    addi a2, a2, %lo(.LCPI12_0)
```

Comment:
We should be able to check the alignment of the constant pool and fold the +4 like we did before.

Reply:
This seems like a useful generic optimization as well.  I assume you're going to follow up here?

Reply:
Yes. I have a patch I'll post later. It only affected the Zdinx lit tests from this patch. Let me know if you have any suggestions on how to hit this issue outside of Zdinx.

Reply:
I guess if this is specific to constant pools (as opposed to global constants), then we'd have to have another lowering which generates a constant and then reads pieces of it.  I'd look in e.g. i128 legalization or something like that, but on reflection, you're right that his is probably fairly narrow.

Reply:
I think there has been discussion in the scalar efficiency sig about when clang/llvm uses constant pools for e.g. 64-bit integers - Zilsd may be another place to revisit whether those should be split before materialization, or put in a constant pool and loaded with `ld`. https://lists.riscv.org/g/sig-scalar-efficiency/topic/110801992 is the discussion, which I note does focus more on 48-bit instructions for materialisation rather than constant pools.